controllers/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -1372,74 +1372,31 @@ func TestDeleteNSTemplateSet(t *testing.T) {
 		require.Empty(t, result)
 	})
 
-	t.Run("delete when there is no finalizer", func(t *testing.T) {
-		// given an NSTemplateSet resource which is being deleted and whose finalizer was already removed
-		nsTmplSet := newNSTmplSet(namespaceName, spacename, "basic", withoutFinalizer(), withDeletionTs(), withClusterResources("abcde11"), withNamespaces("abcde11", "dev", "stage"))
-		r, req, _ := prepareReconcile(t, namespaceName, spacename, nsTmplSet)
-
-		// when a reconcile loop is triggered
-		_, err := r.Reconcile(context.TODO(), req)
-
-		// then
-		require.NoError(t, err)
-		AssertThatNSTemplateSet(t, namespaceName, spacename, r.Client).
-			DoesNotHaveFinalizer() // finalizer was not added and nothing else was done
-	})
-
-	t.Run("NSTemplateSet not deleted until namespace is deleted", func(t *testing.T) {
+	t.Run("NSTemplateSet deletion errors when namespace is not deleted in 1 min", func(t *testing.T) {
 		// given an NSTemplateSet resource and 1 active user namespaces ("dev")
 		nsTmplSet := newNSTmplSet(namespaceName, spacename, "advanced", withNamespaces("abcde11", "dev", "stage"), withDeletionTs(), withClusterResources("abcde11"))
 		nsTmplSet.SetDeletionTimestamp(&metav1.Time{Time: time.Now().Add(-61 * time.Second)})
-		devNS := newNamespace("advanced", spacename, "dev", withTemplateRefUsingRevision("abcde11"))
-		stageNS := newNamespace("advanced", spacename, "stage", withTemplateRefUsingRevision("abcde11"))
+		devNS := newNamespace("advanced", spacename, "dev", withTemplateRefUsingRevision("abcde11"), withFinalizer())
+		stageNS := newNamespace("advanced", spacename, "stage", withTemplateRefUsingRevision("abcde11"), withFinalizer())
 
-		r, fakeClient := prepareController(t, nsTmplSet, devNS, stageNS)
+		r, _ := prepareController(t, nsTmplSet, devNS, stageNS)
 		req := newReconcileRequest(namespaceName, spacename)
 
-		// only add deletion timestamp, but not delete
-		fakeClient.MockDelete = func(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error {
```

Comment:
there is no problem with this test case, I am removing the MockDelete function here because it is not needed anymore. The fakeClient would do only add a deletionTimestamp now since adding the finalizer.